### PR TITLE
Adjust the release checklist

### LINKED ISSUE #8486: Adjust release checklist to adjust for modern reality
### Problem

The release checklist needs some adjustments in the order of how things happen.

### Solution

Update the markdown file accordingly.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jascks**
**Description**:

Adjust the release checklist to better represent the order of how things happen.

**Related issue(s)**:

Fixes #8486

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/checklist/release.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ This checklist verifies a release is rolled out successfully.
 - [ ] Automated Kubernetes deployment to integration successful
 - [ ] Tag release
 
+## Previewnet
```
Comment (xin-hedera) [inline docs/checklist/release.md:18]: To better organize, should put the `previewnet` and `performance` sections before `Generally Available` as sub sections under a new section `Release Candidates`

File: docs/checklist/release.md [hunk 2]
```diff
@@ -29,15 +36,25 @@ This checklist verifies a release is rolled out successfully.
 - [ ] REST API performance tests
 - [ ] Web3 API performance tests
 
+## Generally Available
+
+- [ ] Tag release
```
Comment (xin-hedera) [inline docs/checklist/release.md:41]: tag release is already in the preparation section

File: docs/checklist/release.md (hunk 1 above)
Comment (jascks) [reply docs/checklist/release.md]: I like that. Done.

File: docs/checklist/release.md (hunk 2 above)
Comment (jascks) [reply docs/checklist/release.md]: Removed.

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**